### PR TITLE
fix(compiler): use rootDirs compilerOption to affect genDir layout.

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: tools/compiler_cli/package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-  "name": "angular2-template-compiler",
-  "version": "0.1.9",
+  "name": "@angular/compiler-cli",
+  "version": "0.2.0",
   "description": "Execute angular2 template compiler in nodejs.",
   "main": "index.js",
   "typings": "index.d.ts",
   "bin": {
-     "ng2tc": "./main.js"
+     "ngc": "./main.js"
```
Comment (tbosch) [inline tools/compiler_cli/package.json]: This needs to be updated in the `build.sh` file as well

File: tools/compiler_cli/src/codegen.ts [hunk 2]
```diff
@@ -130,9 +135,16 @@ export class CodeGenerator {
...
+              let root = this.basePath;
+              for (let eachRootDir of this.options.rootDirs) {
+                if (this.ngOptions.trace) console.log(`Check if ${generated.moduleUrl} is under rootDirs element ${eachRootDir}`);
+                if (path.relative(eachRootDir, generated.moduleUrl).indexOf('.') !== 0) {
+                  root = eachRootDir;
+                }
+              }
+
               const emitPath = path.join(this.ngOptions.genDir,
-                                         path.relative(this.basePath, generated.moduleUrl));
```
Comment (tbosch) [inline tools/compiler_cli/src/codegen.ts]: What are we using `basePath` for now?

File: tools/compiler_cli/package.json (hunk 1 above)
Comment (alexeagle) [inline tools/compiler_cli/package.json]: yes, but only when we change shrinkwrap to get a new version of the compiler. Remember that angular currently gets the compiler from an npm dependency to break the cycle.

File: tools/compiler_cli/src/codegen.ts (hunk 2 above)
Comment (alexeagle) [inline tools/compiler_cli/src/codegen.ts]: it's the default for root here - if you don't use `rootDirs` option then your files are relative to location of tsconfig.json

**Comment by alexeagle** [comment]
merged as a033f83

**Pull request closed without merging**